### PR TITLE
gun lockers

### PR DESCRIPTION
**Pull request by dannno**
adds gun lockers
- shotgun variant
- energy gun variant

not added to the map because fuck you

### CODE REVIEW

**Comment by SconesC** [comment]
I've been waiting to see these sprites used for a while.

Shotgun variant holds all different types of shotguns? (Combat, Riot, those older Riot ones, etc.)

**Comment by dannno** [comment]
yuh

File: code/game/objects/structures/guncase.dm [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	icon = 'icons/obj/closet.dmi'
+	icon_state = "gcase-0-open"
+	anchored = 0
+	density = 1
+	opacity = 0
+	var/open = 1
+	var/capacity = 4
+
+/obj/structure/guncase/initialize()
+	for(var/obj/item/I in loc.contents)
```
Comment (Iamgoofball) [inline code/game/objects/structures/guncase.dm]: I'd call ..() just to make sure.

**Comment by Iamgoofball** [comment]
I love it but there's some copy pasta and problems that should be fixed. I can make a pr.

**Comment by Nienhaus** [comment]
These have been sprited for like, over 6 months @SconesC And because it wasn't said I did the sprite, could be redone if wasn't lazy.

File: code/game/objects/structures/guncase.dm [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		else
+			open = 0
+		update_icon()
+
+/obj/structure/guncase/attack_hand(mob/user)
+	if(isrobot(user) || isalien(user))
+		return
+	if(contents.len && open)
+		var/obj/item/weapon/gun/projectile/shotgun/choice = input("Which gun would you like to remove from the case?") as null|obj in contents
+		if(choice)
```
Comment (Iamgoofball) [inline code/game/objects/structures/guncase.dm]: >usr

File: code/game/objects/structures/guncase.dm [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+			if(!usr.canmove || usr.stat || usr.restrained() || !in_range(loc, usr))
```
Comment (Iamgoofball) [inline code/game/objects/structures/guncase.dm]: We need a general proc for this shit.

Comment (RemieRichards) [inline code/game/objects/structures/guncase.dm]: we have some, user.incapacitated() and Adjacent().

Comment (Iamgoofball) [inline code/game/objects/structures/guncase.dm]: Ah

Use this shit then

File: code/game/objects/structures/guncase.dm (hunk 1 above)
Comment (dannno) [inline code/game/objects/structures/guncase.dm]: no

**Comment by dannno** [comment]
ill do it all later then

**Comment by Antimattercarp** [comment]
They lock?
Is this modded bookcase code?

**Comment by dannno** [comment]
no

**Comment by Jordie0608** [comment]
These could really just be resprited lockers couldn't they?

File: html/changelogs/dannno-gunlockers.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+author: Dannno
+
+delete-after: True
+
+  - rscadd: "Added gun lockers for shotguns and energy guns. Click on the lockers with an item to close them if they're full."
```
Comment (Jordie0608) [inline html/changelogs/dannno-gunlockers.yml]: You're missing the 'changes:' header from this

File: tgstation.dme [hunk 5]
```diff
@@ -6,6 +6,49 @@
...
+#define FILE_DIR "sound/machines"
+#define FILE_DIR "sound/mecha"
+#define FILE_DIR "sound/misc"
+#define FILE_DIR "sound/piano"
+#define FILE_DIR "sound/spookoween"
+#define FILE_DIR "sound/violin"
+#define FILE_DIR "sound/voice"
+#define FILE_DIR "sound/voice/complionator"
+#define FILE_DIR "sound/vox_fem"
+#define FILE_DIR "sound/weapons"
```
Comment (Jordie0608) [inline tgstation.dme]: You'll need to turn off 'Automatically set FILE_DIR for sub-directories' in build preferences for DM

**Comment by dannno** [comment]
>These could really just be resprited lockers couldn't they?
no

File: tgstation.dme (hunk 5 above)
Comment (dannno) [inline tgstation.dme]: wut

File: code/game/objects/structures/guncase.dm (hunk 3 above)
Comment (dannno) [inline code/game/objects/structures/guncase.dm]: how does adjacent work

**Comment by dannno** [comment]
k done

**Comment by Jordie0608** [comment]
Why not =|

**Comment by dannno** [comment]
because when you opened it the guns would just by lying around inside it?? and it would look dumb??

**Comment by dannno** [comment]
These are done unless anyone has any criticism or suggestions to improve them.

**Comment by dannno** [comment]
helllo

**Comment by dannno** [comment]
helllloooooooooooo

**Comment by Petethegoat** [comment]
hellooooooooooooooooooooooooooooooooooooooooofart

**Comment by hornygranny** [comment]
i don't like this code

**Comment by dannno** [comment]
please elaborate, how can i improve it

**Comment by dannno** [comment]
@hornygranny I changed it from a choice menu to a popup with html nonsense
that's the best I can do unless someone explains nanoUI to me.

**Comment by dannno** [comment]
@hornygranny @Miauw62 @RemieRichards @Cheridan

**Comment by dannno** [comment]
@hornygranny @Miauw62 @RemieRichards @Cheridan 
They have the proper black UI now

**Comment by dannno** [comment]
@RemieRichards

**Comment by dannno** [comment]
you do me like this now @Luigi?
now you do me like this now?

File: code/game/objects/structures/guncase.dm [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+	update_icon()
+
+/obj/structure/guncase/update_icon()
+	icon_state = "[case_type]-[contents.len]-[open ? "open" : "closed"]"
+
+/obj/structure/guncase/attackby(obj/item/I, mob/user, params)
+	if(isrobot(user) || isalien(user))
+		return
+	if(istype(I, gun_category))
+		if(contents.len < 4 && open)
```
Comment (RemieRichards) [inline code/game/objects/structures/guncase.dm]: replace this 4 with the ```capacity``` var

**Pull request merged**